backend/FwHeadless/Services/ProjectSyncStatusService.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections.Concurrent;
+
+namespace FwHeadless.Services;
+
+public class ProjectSyncStatusService()
+{
+    private ConcurrentDictionary<Guid, ProjectSyncStatus> Status { get; init; } = new();
+
+    public void StartSyncing(Guid projectId)
+    {
+        Status.AddOrUpdate(projectId, (_) => ProjectSyncStatus.Syncing, (_, _) => ProjectSyncStatus.Syncing);
+    }
+
+    public void StopSyncing(Guid projectId)
+    {
+        Status.AddOrUpdate(projectId, (_) => ProjectSyncStatus.ReadyToSync, (_, _) => ProjectSyncStatus.ReadyToSync);
+    }
+
+    public ProjectSyncStatus? SyncStatus(Guid projectId)
+    {
+        return Status.TryGetValue(projectId, out var status) ? status : null;
+    }
+}
+
+public class ProjectSyncStatusUpdater : IDisposable
```

Comment:
I've gotten rid of this in the latest revision, replacing it with a call to Defer.Action since this disposable was only replacing a `try...finally` that called a single method.